### PR TITLE
Add `registerAll*` functions to the Base module

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR exposes the new package entry point `'handsontable/registry'` that allows registering all build-in modules using only one function call. This change is dictated by the needs of the modularized wrappers, which after modularization will depend on the `'handsontable/base'` package not `'handsontable'`. To help developers migrate the new wrappers to the state before modularization there will be possible to register all modules at once instead of registering separately.

PR solution:
```js
import { registerAllModule } from 'handsontable/registry';

registerAllModule();
```
Without the proposition, it would be necessary to import and register desired modules separately. Like in this [modules cheatsheet](https://handsontable.com/docs/modules/#modules-cheatsheet).

The new entry point exposes the following functions:
  * `registerAllEditors` Registers all build-in editors;
  * `registerAllRenderers` Registers all build-in renderers;
  * `registerAllValidators` Registers all build-in validators;
  * `registerAllCellTypes` Registers all build-in cell types;
  * `registerAllPlugins` Registers all build-in plugins;
  * `registerAllModules` Registers all above build-in modules;

### How has this been tested?

I tried implementing the mentioned functions to the Base module, as it would be the best looking approach (`import Handsontable, { registerAllModules } from 'handsontable/base'`) but there was a problem with bundling. The code becomes less tree-shakable in development and production modes. That's why I stuck to the `/registry` entry point.

I tested the changes using Webpack 4 and 5 in different modes.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #8816
2.
3.

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [x] My change requires a change to the documentation.

### CODE REVIEW

**Comment by budnix** [comment]
> Code looks great 🏆
> 
> But I think that `/docs` are missing for all new exports. We should describe them on our modules page, right?

Right! As you accepted the changes, the API becomes stable so with help @kirszenbaum we can start working on docs.

**Review by wojciechczerniak — DISMISSED** [review]
I see docs are being worked on within #8892. No need to block this PR then

File: .changelogs/8819.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "Added support for modularize React wrapper",
```
Comment (kirszenbaum) [inline .changelogs/8819.json:2]:
```suggestion
  "title": "Added support for the modularized React wrapper.",
```

File: .changelogs/8820.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "Added support for modularize Vue wrapper",
```
Comment (kirszenbaum) [inline .changelogs/8820.json:2]:
```suggestion
  "title": "Added support for the modularized Vue wrapper.",
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

**Pull request merged**